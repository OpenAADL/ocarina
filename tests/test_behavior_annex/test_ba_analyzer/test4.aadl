package BA_example1 public
with Base_Types;

data stack
  end stack;

  data implementation stack.default
    subcomponents
      elems : data Base_Types::Integer [100];
      sp : data Base_Types::Integer;
  end stack.default; 


subprogram addition
 features
   x: in parameter Base_Types::Integer;
   y: in parameter Base_Types::Integer;
   r: out parameter Base_Types::Integer;
   ovf: out parameter Base_Types::Boolean;
end addition;

subprogram implementation addition.default
 annex behavior_specification {**

 -- Errors : The variable identifiers must not
 --          conflict with mode identifiers,
 --          feature identifiers, or data
 --          subcomponent identifiers of the component
 --          for which the behavior annex subclause
 --          is defined.(Naming rule N1)


  variables
  v : stack;
  s,y : Stack;
 **};
end addition.default;

end BA_example1;
