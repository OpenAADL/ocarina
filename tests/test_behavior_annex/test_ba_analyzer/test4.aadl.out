test4.aadl:35:05: "y" conflicts with declaration at line 17
